### DISCUSSION TITLE
Method to load SPCA files from existing simulation

### DISCUSSION THREAD

**Discussion by laat0003** [modflowpy/flopy#2020] [Q&A] [answered]
Hi,

I'm loading in an existing simulation originally created with Flopy. My flow model and transport model and their packages all load in fine. The model uses SPCA type files for concentrations via the SSM package. I can access SSM as an object after the simulation is loaded.  What I would like to do is edit the SPCA files and re-run the model but I can't figure out if this is possible. I tried a basic load from file via flopy.mf6.ModflowUtlspca.load() and passing in a path to the SPCA input file but this fails because the load method wants a package object. This seems like a circular reference issue seeing as I'm trying to create the object via a file load but the file load method wants an object. Perhaps I am going about this the wrong way so thought I'd post this as a question in case there is a different way to access SPCA data when loading an existing model.

Thanks in advance for your time,
Tariq

**Comment by spaulins-usgs** [comment, marked as answer]
@langevin-usgs,

I believe spc and spca input files are suppose to behave like lake table input files.  If so, I have a fix for this (see my PR #2025).
I also added test code to autotest since we are currently not testing the spc or spca packages. 

@laat0003,

PR #2025 should fix a problem with loading spc and spca input files, allowing flopy to load them when MFSimulation.load is called.  I will update this thread once it is merged.

**Reply by laat0003** [reply]
@spaulins-usgs 
That is great to hear. Thank you very much. I managed to work around it in the meantime by rebuilding the spca package completely after writing a function to parse the arrays in the original input file.
